### PR TITLE
Allow client to send metrics to IPV6 addresses

### PR DESCRIPTION
**Pull request by Palleas**
The client is currently unable to push metrics to ipv6 addresses, it fails with `sendmsg(descriptor:msgHdr:flags:): Address family not supported by protocol (errno: 97)`, since we bind on 0.0.0.0. 

This pull-request changes the bind instruction to `::` if we're using ipv6 and keeps `0.0.0.0` otherwise.

### CODE REVIEW

File: Tests/StatsdClientTests/TestServer.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftStatsdClient open source project
+//
+// Copyright (c) 2019 the SwiftStatsdClient project authors
```
Comment (tomerd) [inline Tests/StatsdClientTests/TestServer.swift]: nit 2022

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by tomerd** [comment]
@swift-server-bot add to allowlist

**Review by tomerd — APPROVED** [review]
looks good, thanks

File: Tests/StatsdClientTests/StatsdClientIPV6Tests.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import XCTest
+import CoreMetrics
+@testable import StatsdClient
+
+private let port = 9999
+
+class StatsdClientIPV6Tests: XCTestCase {
+
+    func testIPV6Address() throws {
+        let server = TestServer(host: "::1", port: port)
```
Comment (Lukasa) [inline Tests/StatsdClientTests/StatsdClientIPV6Tests.swift:55]: This test can't unconditionally run: some systems don't have IPv6 enabled, and they'll fail. SwiftNIO defends against this with a skip:

```
    static var supportsIPv6: Bool {
        do {
            let ipv6Loopback = try SocketAddress.makeAddressResolvingHost("::1", port: 0)
            return try System.enumerateDevices().filter { $0.address == ipv6Loopback }.first != nil
        } catch {
            return false
        }
    }
```

File: Tests/StatsdClientTests/StatsdClientIPV6Tests.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        XCTAssertNoThrow(try server.connect().wait())
+        defer { XCTAssertNoThrow(try server.shutdown()) }
+        
+        var data = [String]()
+
+        let semaphore = DispatchSemaphore(value: 0)
+        server.onData { _, line in
+            semaphore.signal()
+
+            data.append(line)
```
Comment (Lukasa) [inline Tests/StatsdClientTests/StatsdClientIPV6Tests.swift]: These lines must occur in the opposite order: you have to write to the `data` _first_, and then signal the semaphore.

**Review by Lukasa — CHANGES_REQUESTED** [review]

File: Tests/StatsdClientTests/StatsdClientIPV6Tests.swift (hunk 3 above)
Comment (Palleas) [reply Tests/StatsdClientTests/StatsdClientIPV6Tests.swift]: Good point. I initially forgot the `defer`, but I just removed that part altogether. In this test we don't really care about the line that was written, just that the client was able to connect I guess.

File: Tests/StatsdClientTests/TestServer.swift (hunk 1 above)
Comment (Palleas) [reply Tests/StatsdClientTests/TestServer.swift]: Done!

**Pull request merged**